protocol-units/bridge/chains/ethereum/src/client.rs

```diff
@@ -312,7 +365,7 @@ impl BridgeContractInitiator for EthClient {
 			hash_lock: HashLock(eth_details.hash_lock),
 			//@TODO unit test these wrapping to check for any nasty side effects.
 			time_lock: TimeLock(eth_details.time_lock.wrapping_to::<u64>()),
-			amount: Amount(eth_details.amount.wrapping_to::<u64>()),
+			amount: Amount(AssetType::EthAndWeth((0,eth_details.amount.wrapping_to::<u64>()))),
```

Comment:
Wouldn't the asset always be WETH on `BridgeTransferDetails`? As its what's stored on the contract? No Eth is every stored in the contract.

Reply:
yes, that's why on BridgeTransferDetails we only use the second param for EthAndWeth